src/tir/schedule/primitive/fuse_split.cc

```diff
@@ -0,0 +1,483 @@
+}
+
+StmtSRef Fuse(ScheduleState self, const Array<StmtSRef>& loop_srefs) {
+  //     Invariance
+  //   - The total repeat number has not changed for each direct child block.
+  //   - The execution order has not changed. (The block executes with the same
+  //     args and the same order with before.)
+  std::vector<const ForNode*> loops;
+  loops.reserve(loop_srefs.size());
+  StmtSRef outer_sref{nullptr};
+  const ForNode* outer_loop = nullptr;
+  arith::Analyzer analyzer;
+  // Step 1. check correctness
+  GetScopeRootAndCheckStagePipeline(self, loop_srefs[0]);
+  for (const StmtSRef& sref : loop_srefs) {
+    const auto* loop = sref->StmtAs<ForNode>();
+    if (loop == nullptr) {
+      throw NotLoopError(self->mod, sref->stmt->GetTypeKey());
+    }
+    if (!loop->annotations.empty()) {
+      throw HasAnnotationError(self->mod, GetRef<For>(loop));
+    }
+    if (loop->thread_binding.defined()) {
+      throw HasThreadBindingError(self->mod, GetRef<For>(loop));
+    }
+    if (outer_sref.defined()) {
+      if (sref->parent != outer_sref.get()) {
+        throw OuterNotInnerParent(self->mod, GetRef<For>(outer_loop), GetRef<For>(loop));
+      }
+      Array<Stmt> outer_children = GetChildren(GetRef<Stmt>(outer_loop));
+      if (outer_children.size() != 1 || outer_children[0].get() != loop) {
+        throw NotOnlyChildError(self->mod, GetRef<For>(outer_loop), GetRef<For>(loop));
+      }
+    }
+    outer_sref = sref;
+    outer_loop = loop;
+    if (!analyzer.CanProve(loop->min == 0)) {
+      throw LoopNotStartWithZeroError(self->mod, GetRef<For>(loop));
+    }
+    loops.push_back(loop);
+  }
+  // Step 2. Create fused loop var and replace the original loop vars
+  std::string suffix;
+  for (size_t i = 1; i < loops.size(); i++) {
+    suffix += "_" + loops[i]->loop_var->name_hint;
+  }
+  suffix += "_fused";
+  Var fused_var = loops[0]->loop_var.copy_with_suffix(suffix);
+  Array<PrimExpr> substitute_value;
+  substitute_value.resize(loops.size());
+  PrimExpr tot = fused_var;
+  for (int i = loops.size() - 1; i >= 0; i--) {
+    substitute_value.Set(i, floormod(tot, loops[i]->extent));
+    tot = floordiv(tot, loops[i]->extent);
+  }
+  Stmt loop_body = loops.back()->body;
+  Map<Block, Block> opaque_block_reuse;
+  auto substitute_function = [&](const Var& v) -> Optional<PrimExpr> {
+    for (size_t i = 0; i < loops.size(); i++) {
+      if (v.same_as(loops[i]->loop_var)) {
+        return substitute_value[i];
+      }
+    }
+    return NullOpt;
+  };
+  Stmt new_loop_body =
+      SubstituteAndCollectOpaqueBlock(loop_body, &opaque_block_reuse, substitute_function);
+  // Step 3. Generate a loop to replace the original loops
+  PrimExpr fused_min = 0;
+  PrimExpr fused_extent = 1;
+  for (size_t i = 0; i < loops.size(); i++) {
+    fused_extent *= loops[i]->extent;
+  }
+  fused_extent = analyzer.Simplify(fused_extent);
+  For fused_loop = For(fused_var, fused_min, fused_extent, loops[0]->kind, new_loop_body);
```

Comment:
isn't `loops[0]->kind == ForKind::kSerial`? If so, please replace this with `ForKind::kSerial` instead